community/php7/APKBUILD

```diff
@@ -137,231 +135,270 @@ _build() {
+	local file
+	provides=$_pkgreal-pear
+	replaces=$provides
+	pkgdesc="$pkgdesc (extension and application repository)"
+	depends="$pkgname-dev $pkgname-xml"
 
 	mkdir -p "$subpkgdir"/usr/bin \
 		"$subpkgdir"/etc/$pkgname
 
-	local file; for file in phpize7 pecl pear; do
-		mv "$pkgdir"/usr/bin/$file "$subpkgdir"/usr/bin/
+	mv "$pkgdir"/etc/$pkgname/pear.conf \
+		"$subpkgdir"/etc/$pkgname/ || return 1
+
+	mv "$pkgdir"/usr/share \
+		"$subpkgdir"/usr/ || return 1
+
+	sed -e "s/\$INCARG/\$INCARG -d extension=xml.so/" \
+		 -i "$pkgdir"/usr/bin/pecl || return 1
+
+	for file in pear peardev pecl; do
+		mv "$pkgdir"/usr/bin/$file \
+			"$subpkgdir"/usr/bin/$file$_suffix || return 1
+		sed -e "s:/usr/bin/php:/usr/bin/php$_suffix:g" \
+			-i "$subpkgdir"/usr/bin/$file$_suffix || return 1
+		ln -sf /usr/bin/$file$_suffix \
+			"$subpkgdir"/usr/bin/$file || return 1
 	done
-	mv "$pkgdir"/etc/$pkgname/pear.conf "$subpkgdir"/etc/$pkgname/ || return 1
-	mv "$pkgdir"/usr/share "$subpkgdir"/usr/
 }
 
-common() {
+config() {
+	provides="phpapi=$_apiver"
+	replaces=
+	depends=
 	pkgdesc="$pkgdesc (common config)"
-	depends=""
 
-	mkdir -p "$subpkgdir"/usr
-	mv "$pkgdir"/usr/lib "$subpkgdir"/usr/ || return 1
-	mv "$pkgdir"/etc "$subpkgdir"/ || return 1
+	mkdir -p "$subpkgdir"/etc/$pkgname/conf.d
+
+	install -Dm644 "$builddir"/php.ini-production \
+		"$subpkgdir"/etc/$pkgname/php.ini || return 1
+
+	# exit with an error if some modules were not in subpackages
+	rmdir "$pkgdir"/usr/lib/$pkgname/modules || return 1
 
-	mkdir "$subpkgdir"/etc/$pkgname/conf.d
+	rm -fr "$pkgdir"/etc "$pkgdir"/var "$pkgdir"/usr/lib
 }
 
-phar() {
-	_package_ext || return 1
+_phar() {
+	local file
+	provides=$_pkgreal-phar-phar
+	replaces=$provides
+	pkgdesc="$pkgdesc (archive script)"
+	depends="$pkgname $pkgname-phar"
+
 	mkdir -p "$subpkgdir"/usr/bin
-	mv "$pkgdir"/usr/bin/phar* "$subpkgdir"/usr/bin/
+
+	rm "$pkgdir"/usr/bin/phar || return 1
+	mv "$pkgdir"/usr/bin/phar.phar \
+		"$subpkgdir"/usr/bin/phar.phar$_suffix || return 1
+	for file in phar$_suffix phar phar.phar; do
+		ln -sf /usr/bin/phar.phar$_suffix \
+			"$subpkgdir"/usr/bin/$file || return 1
+	done
 }
 
-_package_ext() {
-	local extname="${subpkgname#$pkgname-}"
-	local extdepends="$(eval "echo \$depends_$extname")"
-	pkgdesc="PHP7 extension: $extname"
-	depends="$depends $extdepends"
+_extension() {
+	local dep
+	local name=${subpkgname#$pkgname-}
+	local index=$(eval echo \$_index_$name)
+	provides="$_pkgreal-$name"
+	replaces=
+	depends=
 
-	case "$extname" in
-	pdo_*) depends="$depends $pkgname-pdo"
-	esac
+	pkgdesc=$(eval echo \$_pkgdesc_$name)
+	: ${pkgdesc:=$(head -n1 "$builddir"/ext/$name/CREDITS)}
+	pkgdesc="PHP$_suffix extension: ${pkgdesc:-$name}"
 
-	# extension dependencies and loading order
-	local elo=00
-	[ -z "$extdepends" ] || elo=01
+	for dep in $(grep -o "_DEP('$name'.*" "$builddir"/ext/$name/config.w32 | sed "s/[',);]//g"); do
+		[ -z "${_extensions##*$dep*}" ] && depends="$depends $pkgname-$dep"
```

Comment:
Actually this might wrong because *nix based build of php should have different deps because it have much more extensions bundled